Source/Lib/Codec/EbEntropyCoding.c

```diff
@@ -4968,70 +4968,37 @@ static void CodeProfileTier(
         // "general_max_12bit_constraint_flag"
         WriteFlagCavlc(
            bitstreamPtr,
-           1);
+           (scsPtr->encoderBitDepth <= EB_12BIT));
 
         // "general_max_10bit_constraint_flag"
-        if(scsPtr->encoderBitDepth <= EB_10BIT || scsPtr->staticConfig.constrainedIntra == EB_TRUE)
-        {
-            WriteFlagCavlc(
-               bitstreamPtr,
-               1);
-        } else
-        {
-            WriteFlagCavlc(
-               bitstreamPtr,
-               0);
-        }
+        WriteFlagCavlc(
+           bitstreamPtr,
+           (scsPtr->encoderBitDepth <= EB_10BIT));
 
         // "general_max_8bit_constraint_flag"
-        //if(scsPtr->encoderBitDepth == EB_8BIT)
-        if(scsPtr->encoderBitDepth == EB_8BIT && (scsPtr->chromaFormatIdc == EB_YUV444 || scsPtr->staticConfig.constrainedIntra == EB_TRUE))
-        {
-            WriteFlagCavlc(
-               bitstreamPtr,
-               1);
-        } else
-        {
-            WriteFlagCavlc(
-               bitstreamPtr,
-               0);
-        }
+        WriteFlagCavlc(
+           bitstreamPtr,
+           (scsPtr->encoderBitDepth <= EB_8BIT && scsPtr->chromaFormatIdc != EB_YUV422));
```

Comment:
Hi @Yeongjin-Jeong , what is the reason of adding "scsPtr->chromaFormatIdc != EB_YUV422"?

Reply:
Hi @tianjunwork, Because there is no profile for 8-bit 4:2:2 in MainRext. So we should obey the constraints to main-422-10 profile.

Reply:
Thx. I find it in line 628 https://hevc.hhi.fraunhofer.de/HM-doc/_t_app_enc_cfg_8cpp_source.html

Reply:
But in TABLE I Profiles Defined in HEVC Version 1 and RExt and the Availability of Features of paper:
https://ieeexplore.ieee.org/document/7265015?arnumber=7265015
main 4:4:4 profile support 8bit and yuv 422.
Except HM, other implementations simple set scsPtr->encoderBitDepth <= EB_8BIT for general_max_8bit_constraint_flag

Reply:
Yes, main-444 also covers 422/8 bits. But in x265 or Intel Meda SDK (when mapping their MainRext to vaapi's MainRext), they used main-422-10 instead of main-444.
(https://github.com/videolan/x265/blob/master/source/encoder/entropy.cpp#L399
https://github.com/Intel-Media-SDK/MediaSDK/blob/master/_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_vaapi.cpp#L46)

Reply:
Hi @Yeongjin-Jeong ,

I think it's better to change with the condition (scsPtr->encoderBitDepth <= EB_8BIT) only to set general_max_8bit_constraint_flag, like what settings do for general_max_12bit_constraint_flag or general_max_10bit_constraint_flag, although the HEVC spec or the [table](https://ieeexplore.ieee.org/mediastore_new/IEEE/content/media/76/7372356/7265015/marpe.t1-2478707-large.gif) doesn't indicate to support 8-bit 4:2:2.

Reply:
Hi @Austin-Hu, Thanks for your suggestion.
If there is h265 parser that does not allow not standard constraints, It may be a problem. But most h265 parser(e.g. gstreamer) seem to match closet profile if bitstream contains not standard constraints. So I applied your review to this PR. :)